src/runtime/storage/firebase-storage.ts

```diff
@@ -1467,3 +1477,131 @@ class FirebaseBigCollection extends FirebaseStorageProvider implements BigCollec
     throw new Error('unimplemented');
   }
 }
+
+/**
+ * Thin wrapper providing a Collection API for a Firebase reference. This is used by
+ * FirebaseVariable and FirebaseCollection to access their underlying backing store
+ * collections without pulling the entirety of those collections locally.
+ */
+class FirebaseBackingStore extends FirebaseStorageProvider implements CollectionStorageProvider {
+  public maxConcurrentRequests = 5;
+
+  private childRef(id) {
+    return this.reference.child('items/' + FirebaseStorage.encodeKey(id));
+  }
+
+  async store(value, keys: string[]) {
+    await this.storeSingle(value, keys);
+  }
+
+  async storeMultiple(values, keys: string[]) {
+    while (values.length > 0) {
+      const chunk = values.splice(0, this.maxConcurrentRequests);
+      await Promise.all(chunk.map(value => this.storeSingle(value, keys)));
+    }
+  }
+
+  private storeSingle(value, keys: string[]) {
+    return this.childRef(value.id).transaction(data => {
+      if (data === null) {
+        data = {value, keys: {}};
+      } else {
+        // Allow legacy mutation for now.
+        data.value = value;
+      }
+      for (const key of keys) {
+        const encKey = FirebaseStorage.encodeKey(key);
+        data.keys[encKey] = 1;
+      }
+      return data;
+    }, undefined, false);
+  }
+
+  async remove(id: string, keys: string[]) {
+    await this.removeSingle(id, keys);
+  }
+
+  async removeMultiple(items) {
+    if (items.length === 0) {
+      await this.reference.child('items').remove();
+    } else {
+      while (items.length > 0) {
+        const chunk = items.splice(0, this.maxConcurrentRequests);
+        await Promise.all(chunk.map(item => this.removeSingle(item.id, item.keys)));
+      }
+    }
+  }
+
+  private removeSingle(id: string, keys: string[]) {
+    return this.childRef(id).transaction(data => {
+      if (data === null) {
+        return null;
+      }
+      if (keys.length > 0) {
+        keys.forEach(key => delete data.keys[FirebaseStorage.encodeKey(key)]);
+        if (Object.keys(data.keys).length > 0) {
+          return data;
+        }
+      }
+      // Returning an empty object deletes the Firebase node.
+      return {};
+    }, undefined, false);
+  }
+
+  async get(id: string) {
+    const snapshot = await this.childRef(id).once('value');
```

Comment:
could you not .get() here to register for mutations?

Reply:
Yes, if you want to view mutations on a specific existing entity, we can attach a listener for that. For a referenceMode collection to observe mutations to any of its items, we'll need to attach a listener for each item in the collection. No idea if that has performance implications for the Firebase servers.

Reply:
OK, got it. Performance considerations aside, would that fix the break you described? Or do I misunderstand the problem?

Reply:
Should do. I'll update the PR and run some tests to verify.

Reply:
Thanks!